apps/prairielearn/src/lib/regrading.js

```diff
@@ -5,101 +5,93 @@ import * as path from 'path';
-        user_id: user_id,
-        authn_user_id: authn_user_id,
-        type: 'regrade_assessment_instance',
-        description: 'Regrade ' + assessment_instance_label + ' for ' + jobInfo,
-        job_sequence_id: job_sequence_id,
-        last_in_sequence: true,
-      };
-      createJob(jobOptions, function (err, job) {
-        if (err) {
-          logger.error('Error in createJob()', err);
-          failJobSequence(job_sequence_id);
-          return;
-        }
-        job.verbose('Regrading ' + assessment_instance_label + ' for ' + jobInfo);
-        var params = [assessment_instance_id, authn_user_id];
-        sqldb.call('assessment_instances_regrade', params, function (err, result) {
-          if (ERR(err, function () {})) {
-            job.fail(err);
-          } else {
-            if (result.rowCount !== 1) {
-              job.fail(new Error('Incorrect rowCount: ' + result.rowCount));
-            }
-            job.verbose('Regrading complete');
-            var regrade = result.rows[0];
-            if (regrade.updated) {
-              job.verbose('Questions updated: ' + regrade.updated_question_names.join(', '));
-              job.verbose(
-                'New score: ' +
-                  Math.floor(regrade.new_score_perc) +
-                  '% (was ' +
-                  Math.floor(regrade.old_score_perc) +
-                  '%)',
-              );
-            } else {
-              job.verbose('No changes made');
-            }
-            ltiOutcomes.updateScore(assessment_instance_id, (err) => {
-              if (err) {
-                job.fail(err);
-              } else {
-                job.succeed();
-              }
-            });
-          }
-        });
-      });
-    });
-  });
+  var jobOptions = {
+    course_id: course_id,
+    course_instance_id: course_instance_id,
+    assessment_id: assessment_id,
+    user_id: user_id,
+    authn_user_id: authn_user_id,
+    type: 'regrade_assessment_instance',
+    description: 'Regrade ' + assessment_instance_label + ' for ' + jobInfo,
+    job_sequence_id: job_sequence_id,
+    last_in_sequence: true,
+  };
+  const job = await createJobAsync(jobOptions);
+  job.verbose('Regrading ' + assessment_instance_label + ' for ' + jobInfo);
+  var params = [assessment_instance_id, authn_user_id];
+  const jobResult = await sqldb.callAsync('assessment_instances_regrade', params);
+  if (jobResult.rowCount !== 1) {
+    job.fail(new Error('Incorrect rowCount: ' + jobResult.rowCount));
+  }
+  job.verbose('Regrading complete');
+  var regrade = jobResult.rows[0];
+  if (regrade.updated) {
+    job.verbose('Questions updated: ' + regrade.updated_question_names.join(', '));
+    job.verbose(
+      'New score: ' +
+        Math.floor(regrade.new_score_perc) +
+        '% (was ' +
+        Math.floor(regrade.old_score_perc) +
+        '%)',
+    );
+  } else {
+    job.verbose('No changes made');
+  }
+  ltiOutcomes.updateScoreAsync(assessment_instance_id);
```

Comment:
This is an async function; it should be awaited.

Reply:
This is now awaited in commit [3dc5d16](https://github.com/PrairieLearn/PrairieLearn/pull/9080/commits/3dc5d16ccba40aec11a370c40ace6725c5f72c21).